autogen/agentchat/contrib/functions/file_utils.py

```diff
@@ -0,0 +1,220 @@
+from typing import Optional
+from .functions_utils import FunctionWithRequirements
+
+
+@FunctionWithRequirements(python_packages=["pdfminer.six", "requests"])
+def read_text_from_pdf(file_path: str) -> str:
+    """
+    Reads text from a PDF file and returns it as a string.
+
+    Args:
+        file_path (str): The path to the PDF file.
+
+    Returns:
+        str: The extracted text from the PDF file.
+    """
+    import io
+    import requests
+    from pdfminer.high_level import PDFResourceManager, PDFPageInterpreter
+    from pdfminer.converter import TextConverter
+    from pdfminer.pdfpage import PDFPage
+
+    resource_manager = PDFResourceManager()
+    text_stream = io.StringIO()
+    converter = TextConverter(resource_manager, text_stream)
+    interpreter = PDFPageInterpreter(resource_manager, converter)
+
+    if file_path.startswith("http://") or file_path.startswith("https://"):
+        response = requests.get(file_path)
+        file = io.BytesIO(response.content)
+    else:
+        file = open(file_path, "rb")
+
+    for page in PDFPage.get_pages(file):
+        interpreter.process_page(page)
+
+    text = text_stream.getvalue()
+    converter.close()
+    text_stream.close()
+
+    return text
+
+
+@FunctionWithRequirements(python_packages=["python-docx"])
+def read_text_from_docx(file_path: str) -> str:
+    """
+    Reads text from a DOCX file and returns it as a string.
+
+    Args:
+        file_path (str): The path to the DOCX file.
+
+    Returns:
+        str: The extracted text from the DOCX file.
+    """
+    from docx import Document
+
+    doc = Document(file_path)
+    paragraphs = [p.text for p in doc.paragraphs]
+    text = "\n".join(paragraphs)
+
+    return text
+
+
+@FunctionWithRequirements(python_packages=["pillow", "requests", "easyocr"])
```

Comment:
EasyOCR involved installing torch (which also involves installing many Nvida dependencies). This is multi-gigabytes amd takes forever. I feel like we should handle such large dependencies differently.